### PR TITLE
*: Adds unithash to status output

### PR DESCRIPTION
**Pull request by zeisss**
Fixes https://github.com/giantswarm/giantswarm/issues/524
1. This PR adds a `UnitHash` field to the `MachineState` based on the `Hash` of units in fleet. This is returned for the `Controller.GetStatus` calls. 
2. The CLI `formicactl status <group>` command was changed to take the `UnitHash` field into account for grouping as well. Added a test for this too.

```
core@core-01 ~ $ ./formicactl status demo
Group   Units  FDState   FCState   SAState  IP            Machine

demo@1  *      launched  launched  active   172.17.8.103  99f23dbedf2f4bdd9ed9f2600b5944c3
demo@2  *      launched  launched  active   172.17.8.101  4ea5734c9d6347829d9236502b79c6ca

core@core-01 ~ $ ./formicactl status demo -v
Group   Units                FDState   FCState   SAState  Hash                                      IP            Machine

demo@1  demo-main@1.service  launched  launched  active   fa59254bb1fac86a10935d9aaf839fe0638fbaba  172.17.8.103  99f23dbedf2f4bdd9ed9f2600b5944c3
demo@2  demo-main@2.service  launched  launched  active   fa59254bb1fac86a10935d9aaf839fe0638fbaba  172.17.8.101  4ea5734c9d6347829d9236502b79c6ca
```

### CODE REVIEW

**Comment by zeisss** [comment]
ping @giantswarm/cena

File: controller/status_test.go [hunk 1]
```diff
@@ -6,9 +6,61 @@ import (
 	"reflect"
 	"testing"
 
+	. "github.com/onsi/gomega"
+
 	"github.com/giantswarm/formica/fleet"
 )
 
+func givenSingleUnitStatus(name, sliceID string) fleet.UnitStatus {
```
Comment (xh3b4sd) [inline controller/status_test.go:14]: Where is this naming coming from? I notice you prefix your helper methods with `given`.

**Comment by xh3b4sd** [comment]
The current state of this PR does not fix the given issue IMO. The hash is not displayed when fetching the status. I have expected changes in the cli package. There are none. What do I miss?

**Comment by zeisss** [comment]
> What do I miss?

A hammer to hit me. You are correct.

File: cli/common.go [hunk 2]
```diff
@@ -131,16 +134,12 @@ func createRequest(slices []string) (controller.Request, error) {
 }
 
 var (
-	statusHeader = "Group | Units | FDState | FCState | SAState | IP | Machine"
-	statusBody   = "%s | %s | %s | %s | %s | %s | %s"
+	statusHeader = "Group | Units | FDState | FCState | SAState {{if .Verbose}}| Hash{{end}} | IP | Machine"
```
Comment (zeisss) [inline cli/common.go:137]: I initially tried having two separate variables for the verbose case, but I felt that is ugly too. So I did the loop to go templates.

File: controller/status_test.go (hunk 1 above)
Comment (zeisss) [inline controller/status_test.go:14]: Not sure, I picked it up at TT-S I guess. `given` prefixes more or less helper functions to construct objects in a given state, like a fixture but also for e.g. service objects, not just data objects.

**Comment by zeisss** [comment]
ping @xh3b4sd I added the output.

File: cli/common.go (hunk 2 above)
Comment (xh3b4sd) [inline cli/common.go:137]: Yeah this is tricky. Good enough.

**Comment by xh3b4sd** [comment]
LGTM

File: cli/common.go (hunk 2 above)
Comment (zeisss) [inline cli/common.go:137]: :bow:

**Comment by xh3b4sd** [comment]
I noticed one thing. Sorry to being late though. We extended the status display but we did not test `createStatus`. See https://github.com/giantswarm/formica/pull/46/files#diff-84398aaa4cb21783dbbf26c2c9a4ceb6R142

Would you mind doing that?

**Comment by zeisss** [comment]
Agreed. We added an issue for this in the planning.

**Pull request merged**